python-package/src/atls/utils/requests/adapter.py

```diff
@@ -0,0 +1,110 @@
+import socket
+from typing import Any, Dict, List, Optional, Tuple
+
+from atls import ATLSContext, HTTPAConnection
+from atls.validators import Validator
+from requests.adapters import (
+    DEFAULT_POOLBLOCK,
+    DEFAULT_POOLSIZE,
+    DEFAULT_RETRIES,
+    HTTPAdapter,
+)
+from urllib3 import HTTPSConnectionPool
+from urllib3.poolmanager import PoolManager
+from urllib3.util.retry import Retry as Retry
+
+
+class _HTTPAConnectionShim(HTTPAConnection):
```

Comment:
Does this class need to be defined twice?

Reply:
If you look closely, they're slightly different.

Reply:
Why can't we use this definition for the inner class in the other spot? The only difference I can see is where validators is gotten from